### PR TITLE
feat(react-ui-kit): add DropFileInput component (#SQSERVICES-758)

### PR DESCRIPTION
**Pull request by PatrykBuniX**
Added file input with drop zone area:

- user is able to upload files by dropping files in the zone or by clicking on input's label text (eg. select file from your device)
- customisable:
    - dropdown text (heading and clickable label) 
    - description under input
    - zone's styles (css object)
- `onFilesUploaded` callback passed as props run after file is successfully uploaded using dropdown or native file select; takes an array of files as a first (and only) param.
- `onInvalidFilesDropError` - native file picker won't allow user to pick file with unexpected type, but we have to validate types of files uploaded using drop area. This callback will run after invalid file was dropped on drop area.
- `accept` (native file input's attribute) - accepted file mime-types list joined with comma (`,`)
- `multiple` (native file input's attribute) - allows to upload multiple files
- component supports theme colors

How to test: 
- navigate to `wire-web-packages/packages/react-ui-kit` directory
- run `yarn start`
- http://localhost:8090 -> `DropFileInput` section 

//todo: unit tests

## Pull Request Checklist

- [ ] My code is covered by tests
- [ ] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

File: packages/react-ui-kit/src/Form/DropFileInput.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Demo:
```
Comment (tlebon) [inline packages/react-ui-kit/src/Form/DropFileInput.md:1]: do these md files need a copyright also?

File: packages/react-ui-kit/src/Form/DropFileInput.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+```js
+import {Fragment} from 'react';
```
Comment (tlebon) [inline packages/react-ui-kit/src/Form/DropFileInput.md:4]: is this referenced/necessary?

File: packages/react-ui-kit/src/Form/DropFileInput.md (hunk 1 above)
Comment (PatrykBuniX) [reply packages/react-ui-kit/src/Form/DropFileInput.md:1]: I'm not sure, none of them have one

File: packages/react-ui-kit/src/Form/DropFileInput.md (hunk 2 above)
Comment (PatrykBuniX) [reply packages/react-ui-kit/src/Form/DropFileInput.md:4]: nope, deleting it

File: packages/react-ui-kit/src/Form/DropFileInput.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    headingText,
+    description,
+    accept,
+    multiple,
+    ...inputProps
+  } = props;
+
+  const [isDraggedOver, setIsDraggedOver] = useState(false);
+
+  const handleDragOver = (e: React.DragEvent<HTMLDivElement>) => {
```
[the remark this answers is not in the document]
Comment (PatrykBuniX) [reply packages/react-ui-kit/src/Form/DropFileInput.tsx:117]: `e` for evants is pretty common use and pretty straight forward for me, but I'm ok with `event`also - it's more readable 😄

File: packages/react-ui-kit/src/Form/DropFileInput.test.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+describe('"DropFileInput"', () => {
+  it('matches snapshot', () => matchComponent(<ThemedDropFileInput {...getDefaultProps()} />));
+
+  it('returns file on native file upload', () => {
+    const file = new File(['(⌐□_□)'], 'chucknorris.png', {type: 'image/png'});
+
+    const props = {
+      ...getDefaultProps(),
+      accept: undefined,
+    };
```
Comment (tlebon) [inline packages/react-ui-kit/src/Form/DropFileInput.test.tsx:53]: i dont necessarily understand why you chose to use a function for the default props and why you don't just spread them out in each component and add the necessary accept value. this seems like a little extra than is necessary

Comment (tlebon) [inline packages/react-ui-kit/src/Form/DropFileInput.test.tsx:53]: i think if the props are this simple you can just spread in the component

Comment (PatrykBuniX) [reply packages/react-ui-kit/src/Form/DropFileInput.test.tsx:53]: I need to access props.method later, that's why it' like this

Comment (tlebon) [reply packages/react-ui-kit/src/Form/DropFileInput.test.tsx]: makes sense

File: packages/react-ui-kit/src/Form/DropFileInput.test.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+const xlsxFile = new File(['(⌐□_□)'], 'chucknorris.xlsx', {type: '.xlsx'});
+
+describe('"DropFileInput"', () => {
+  it('matches snapshot', () =>
+    matchComponent(<ThemedDropFileInput {...getDefaultProps({accept: 'image/png, image/jpeg'})} />));
+
+  it('returns file on native file upload', () => {
+    const props = getDefaultProps();
+
+    const {getByTestId} = render(<ThemedDropFileInput {...props} />);
```
[the remark this answers is not in the document]
Comment (PatrykBuniX) [reply packages/react-ui-kit/src/Form/DropFileInput.test.tsx:58]: I can't spread it, I need to access props.onFilesUploaded after rendering

**Pull request merged**